spec/requests/locations_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'rails_helper'
+
+RSpec.describe 'Locations API', type: :request do
+  let!(:locations) { create_list(:location, 10) }
+  let(:location_id) { locations.first.id }
+
+  # GET
+  describe 'Get /locations' do
+    before { get '/locations' }
+
+    it 'returns locations' do
+      expect(json).not_to be_empty
+      expect(json.size).to eq(10)
+    end
+
+    it 'returns status code 200' do
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe 'Get /locations/id' do
+    before do
+      location = Location.find(location_id)
+      location.residents << build(:character)
+      get "/locations/#{location_id}"
```

Comment:
Adding residents manually rather than through factory due to recursive calling between Location and Character. When I added a trait to the location factory to generate residents, it meant when building location, it would then call the character factory which would then generate location object for origin and location. Would end up with the following
Location => generates character 
Character => generates location for origin and location

So each location generated would generate 3 locations. This meant when I generated a list of 10 locations, for the get locations test above, it would actually generate 30. Thought it might look a bit confusing if generate list of 10 locations, but I specify the expected size to be 30.